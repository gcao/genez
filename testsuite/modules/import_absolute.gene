#!/usr/bin/env gene run

# Test absolute module imports

# Test 1: Absolute path imports (from root)
(import "/std/math")
(import "/std/io" :as io)
(import "/std/collections" [List Dict])

# Test 2: System modules (hypothetical standard library)
(import "/sys/os")
(import "/sys/fs" :as filesystem)

# Test 3: Application root imports
(import "/app/config")
(import "/app/models/user" :as User)
(import "/app/controllers/auth" [login logout])

# Test 4: Mixed absolute imports
(import "/lib/vendor/json")
(import "/lib/vendor/http" :as http)
(import "/lib/internal/cache" [Cache MemCache])

(print "Absolute import syntax parsed successfully")