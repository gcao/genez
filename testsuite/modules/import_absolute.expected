Absolute import syntax parsed successfully